### PR TITLE
Course-View

### PR DESCRIPTION
**Pull request by DarrenZhang01**
The router for the different courses

### CODE REVIEW

File: js/component-courses.js [hunk 1]
```diff
@@ -1,2 +1,40 @@
...
+// let courses = '<ul id="app-course">' +
+//         '<li v-for="course in list">{{ course }}</li>' +
+//         '</ul>';
+// let list;
+
+// /* Fetch the returned json object value. */
+// fetch('get_info.php').then(function (response) {
+//     list = JSON.parse(response);
+// });
+
```
Comment (siefkenj) [inline js/component-courses.js]: Don't leave commented out code in PRs

File: js/component-courses.js [hunk 2]
```diff
@@ -1,2 +1,40 @@
...
 // export as global since we're not using a module system
-window.Courses = { template: '<div>Classes</div>' }
+window.Courses = Vue.component('Courses', {
+    name: "Courses",
+    data: function () {
+        return {list:null};
```
Comment (siefkenj) [inline js/component-courses.js]: Run your code through the prettier.js code formatter with indentation at 4 spaces.

File: js/component-courses.js [hunk 3]
```diff
@@ -1,2 +1,40 @@
...
+    },
+    beforeRouteEnter (to, from, next) {
+        let list;
```
Comment (siefkenj) [inline js/component-courses.js]: Where is `list` used?

File: js/component-courses.js [hunk 4]
```diff
@@ -1,2 +1,40 @@
...
+        // console.log("123");
+        /* Fetch the returned json object value. */
+        fetch('get_info.php').then(function (response) {
+            self.list = response.json().then(function (data) {
```
Comment (siefkenj) [inline js/component-courses.js]: Do you handle errors?

**Review by siefkenj — CHANGES_REQUESTED** [review]

File: js/component-courses.js (hunk 1 above)
Comment (DarrenZhang01) [reply js/component-courses.js]: Thanks, I've already removed them in the new commit!

File: js/component-courses.js (hunk 2 above)
Comment (DarrenZhang01) [reply js/component-courses.js]: Thanks, I started using prettier!

File: js/component-courses.js (hunk 3 above)
Comment (DarrenZhang01) [reply js/component-courses.js]: I see there is a redundant there, so I revised it!

File: js/component-courses.js (hunk 4 above)
Comment (DarrenZhang01) [reply js/component-courses.js]: Add it!

File: js/component-courses.js [hunk 5]
```diff
@@ -1,2 +1,36 @@
...
+            err: null
+        };
+    },
+    /* Before this view is rendered, fetch the data from the php file */
+    beforeRouteEnter(to, from, next) {
+        /* Fetch the returned json object value. */
+        fetch("get_info.php")
+            .then(function(response) {
+                response.json().then(function(data) {
+                    next(function(vm) {
```
Comment (siefkenj) [inline js/component-courses.js]: I don't understand this. Where is `next` defined?

File: js/component-courses.js [hunk 6]
```diff
@@ -1,2 +1,36 @@
...
+                        vm.list = data.DATA;
```
Comment (siefkenj) [inline js/component-courses.js]: Check the type of `data` before assigning it to a list. Handle the wrong type by throwing the appropriate error.

**Review by siefkenj — CHANGES_REQUESTED** [review]

File: js/component-courses.js [hunk 7]
```diff
@@ -1,2 +1,41 @@
...
+
+    /* Before this view is rendered, fetch the data from the php file */
+    mounted () {
+        /* Fetch the returned json object value. */
+        fetch("get_info.php")
+        .then(function (response) {
+            return response.json()
+        })
+        .then(data => {
+            if (Array.isArray(data.DATA)) {
```
Comment (siefkenj) [inline js/component-courses.js]: This is not what I meant...you should check if `data.TYPE=="courses"` or if `data.TYPE=="error"` or something else.

File: js/component-courses.js [hunk 8]
```diff
@@ -1,2 +1,41 @@
...
+    data: function() {
+        return {
+            list: null,
+            err: null,
+            warn: false
+        };
+    },
+
+    /* Before this view is rendered, fetch the data from the php file */
+    mounted () {
```
Comment (siefkenj) [inline js/component-courses.js]: `mounted: function() {`

File: js/component-courses.js [hunk 9]
```diff
@@ -1,2 +1,41 @@
...
+                console.log("data.DATA is not an appropriate array!");
+                this.warn = true;
+            }
+        })
+        .catch(error => {
+            this.err = error.toString();
+            console.log("Here is error in fetching data!");
+        });
+    },
+    template: `<div id="app-course">
```
Comment (siefkenj) [inline js/component-courses.js]: Be consistent with spacing. If there are newlines between other attributes, but a newline before them all.

File: js/component-courses.js [hunk 10]
```diff
@@ -1,2 +1,52 @@
...
+            err: null,
+            warn: false
+        };
+    },
+
+    /* Before this view is rendered, fetch the data from the php file */
+    mounted: function () {
+        /* Fetch the returned json object value. */
+        fetch("get_info.php")
+        .then(function (response) {
```
Comment (siefkenj) [inline js/component-courses.js]: Use an arrow function here

File: js/component-courses.js [hunk 11]
```diff
@@ -1,2 +1,52 @@
...
+            // If the data type is courses as expected, set the list attribute
+            // in this component
+            if (data.TYPE == "courses") {
+                console.log(data.DATA);
+                this.list = data.DATA;
+            }
+            // If the data type is error, display the error message
+            else if (data.TYPE == "error") {
+                console.log("data.DATA is not an appropriate array!");
+                this.err = "The error data is getting back from the php file!";
```
Comment (siefkenj) [inline js/component-courses.js]: This is not a good description of the error...

File: js/component-courses.js [hunk 12]
```diff
@@ -1,2 +1,52 @@
...
+        .then(data => {
+            // If the data type is courses as expected, set the list attribute
+            // in this component
+            if (data.TYPE == "courses") {
+                console.log(data.DATA);
+                this.list = data.DATA;
+            }
+            // If the data type is error, display the error message
+            else if (data.TYPE == "error") {
+                console.log("data.DATA is not an appropriate array!");
```
Comment (siefkenj) [inline js/component-courses.js]: You don't know that...You also print an error in another way, so no need for this. There is `console.warn` and `console.error` if you really want to send an error to the console.

**Review by siefkenj — COMMENTED** [review]
Almost there!

**Pull request merged**